### PR TITLE
Datatypes

### PR DESCRIPTION
**Pull request by kiniv**
This is change to improve Q datatypes support.

1) Adding support for timestamp and timespan
2) Fixing few issues where plugin highlight as correct "numbers" incorrect q expressions.
Examples:
Q can have something like 1i, 1f. At the same time 1.5i, 1d or 1m is not allowed.
Also it's possible to have 1.5f, 1e7f and 1.5p, but 1e7p is not allowed again. The list can be extended.
Same issue is with lists. That was fixed to the best of my abilities
3) Expression 1. or .9 are valid floats. Now fixed.
4) Variables in q must start from alphabetic character. In the namespaces where we start with . next character must still be alphabetical. That was fixed. (.2f is now number, not user identifier)

Fixed unit tests accordingly.

Null in q can't be negative, but one of the K tests demand -0N. I hence left it in the plugin, but would like to understand if really need it.

@a2ndrade Can you please review

### CODE REVIEW

**Comment by a2ndrade** [comment]
Thanks for the PR! You can remove the test for `-0N`as it's only valid in k3. In general, there are still a few pieces that remain from k3's support (most of the K tests originally referred to k3, not k4). Besides a small naming change (see comment) and the change for `-0N`, everything else looks good.

**Review by a2ndrade — CHANGES_REQUESTED** [review]
minor naming changes

**Comment by a2ndrade** [comment]
Also, after you make the few updates, could you squash the commits into 1. Thanks!

**Comment by kiniv** [comment]
@a2ndrade Thanks for the feedback. I can't figure out how can I see what naming changes I should do. 
Should I close this PR and raise another with single commit?

File: src/main/resources/com/appian/intellij/k/k.flex [hunk 1]
```diff
@@ -47,28 +47,51 @@ Q_SYSTEM_FUNCTION=(abs|acos|aj|aj0|all|and|any|asc|asin|asof|atan|attr|avg|avgs|
...
 // all numbers
-NUMBER_NO_TYPE=-?({NUMBER_POSITIVE_NEGATIVE})|{NUMBER_POSITIVE}|-?({Q_DATE}|{Q_TIME}|{Q_DATETIME}|{Q_MONTH})
-NUMBER={NUMBER_NO_TYPE}{TYPE}?
+NUMBER={NUMBER_INTEGER}{INT_TYPE}?|({NUMBER_FLOAT}|{NUMBER_EXP}){FLOAT_TYPE}?|
+       ({INFINITY}|{Q_NIL})({INT_TYPE}|{Q_DATETIME_TYPE})?|{Q_NIL}"g"|
+       ({Q_TIMESTAMP}|{Q_TIMESPAN}|{NUMBER_FLOAT}){TIMESPAN_STAMP_TYPE}?|
+       {NUMBER_POSITIVE}|({Q_MINUTE}|{Q_SECOND}|{Q_TIME}){Q_TIME_TYPE}?|{Q_DATE}[dz]?|{Q_DATETIME}"z"?|{Q_MONTH}"m"?
 BINARY_VECTOR=[01][01]+"b"|"0x"{Q_HEX_CHAR}{2}{Q_HEX_CHAR}+
-NUMBER_VECTOR={NUMBER_NO_TYPE}({WHITE_SPACE}{NUMBER_NO_TYPE})+{TYPE}?|{BINARY_VECTOR}
+VECTOR_INT={NUMBER_INTEGER}|{Q_NIL}|{INFINITY}
```
Comment (a2ndrade) [inline src/main/resources/com/appian/intellij/k/k.flex]: @kiniv (I thought I had added a comment here but I probably forgot to submit it), the naming change was around the rules of the form `VECTOR_{TYPE}`. Could we rename them to be `{TYPE}_VECTOR_ITEM` instead? e.g.
`VECTOR_INT` --> `INT_VECTOR_ITEM`
`VECTOR_FLOAT` --> `FLOAT_VECTOR_ITEM`

**Comment by a2ndrade** [comment]
You don't need to create a new PR. Just squash your commits and force push to your branch. The PR will be updated. e.g.
```bash
git rebase -i origin/master
```
select `r(eword)` for the first commit and `s(quash)` for the rest. Save and exit. After the rebase, just force push into your branch
```bash
# will force push the checkout branch (e.g. datatypes)
git push -f original HEAD
```

**Comment by a2ndrade** [comment]
@kiniv would you be OK with me making the changes and pushing the code? I may have some time tomorrow. I'd like to get your changes in soon. Let me know, Thanks!

**Comment by kiniv** [comment]
@a2andrade apologies for delay. Will do shortly

**Comment by kiniv** [comment]
@a2ndrade I did the requested change. Would you like me to squash commits before you merge it or that was just going forward?

**Comment by kiniv** [comment]
@a2ndrade I did the requested change.

**Review by a2ndrade — APPROVED** [review]

**Comment by a2ndrade** [comment]
@kiniv thanks! Looks great. Merging now.

**Pull request merged**